### PR TITLE
[RLlib; Docs overhaul] Docstring cleanup: rllib/utils

### PR DESCRIPTION
**Pull request by sven1977**
Docstring cleanup: rllib/utils folder.

Cleanup docstrings.
- Remove type hints in Args list (already in the signature).
- Remve type hint from "Returns" (already in signature).
- Add docstrings where missing.
- Re-sort some class methods (by their importance, private/public, deprecated, etc..).
- soft-moved/renamed some modules, such as tf_ops -> tf_utils, memory.py's functions moved into numpy.py, etc..

## Checks

- [x] I've run `scripts/format.sh` to lint the changes in this PR.
- [x] I've included any doc changes needed for https://docs.ray.io/en/master/.
- [x] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [x] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

File: rllib/utils/framework.py [hunk 1]
```diff
@@ -258,8 +258,8 @@ def get_variable(value: Any,
 
 
 @Deprecated(
-    old="rllib/models/utils.py::get_activation_fn",
-    new="rllib/utils/framework.py::get_activation_fn",
+    old="rllib/utils/framework.py::get_activation_fn",
+    new="rllib/models/utils.py::get_activation_fn",
```
Comment (gjoliver) [inline rllib/utils/framework.py:262]: did we really swapped new and old 😆

File: rllib/utils/memory.py [hunk 2]
```diff
@@ -1,73 +1,11 @@
...
-            batch_dim = sum(s.shape[0] for s in items)
-            new_shape = (batch_dim, ) + items[0].shape[1:]
-        output = flat.reshape(new_shape)
-        assert output.ctypes.data % 64 == 0, output.ctypes.data
-        np.concatenate(items, out=output, axis=1 if time_major else 0)
-        return output
-    else:
-        return np.concatenate(items, axis=1 if time_major else 0)
+aligned_array = aligned_array
+concat_aligned = concat_aligned
```
Comment (gjoliver) [inline rllib/utils/memory.py:11]: wait, I don't get the point of these 2 lines ...

**Review by gjoliver — COMMENTED** [review]
Just 1 very quick question.

File: rllib/utils/framework.py (hunk 1 above)
Comment (sven1977) [reply rllib/utils/framework.py:262]: haha, yeah, the args were wrong in the deprecation warning :/

File: rllib/utils/memory.py (hunk 2 above)
Comment (sven1977) [reply rllib/utils/memory.py]: It's basically to trick the linter to not complain that we import aligned_array and concat_aligned but are not really using it.
But you are right, I think we can fix that with `#noqa`. ...

**Review by gjoliver — APPROVED** [review]
I see. that makes sense. :)
awesome PR.

File: rllib/utils/memory.py (hunk 2 above)
Comment (sven1977) [reply rllib/utils/memory.py]: done

**Pull request merged**